File: 프로그래머스/정수제곱근판별.cpp
#include <string>
#include <vector>
#include <cmath>
#include <iostream>

using namespace std;

long long solution(long long n) {
	long long answer = 0;
	long long num = sqrt(n);

	if (num * num == n) {
		answer = (num + 1)  * (num + 1);
	}
	else {
		answer = -1;
	}
	return answer;
}

int main() {
	cout << solution(121);
}

File: 프로그래머스/정수제곱근판별.py
def solution(n):
    import math
    num = math.sqrt(n)

    if math.sqrt(n) == int(math.sqrt(n)):
        return int(pow(num+1, 2))
    else:
        answer = -1


    #     return int(pow(num + 1, 2))

    return answer


print(solution(121))
